packages/tenant-process/src/services/tenantService.ts

```diff
@@ -352,4 +475,33 @@ export function tenantServiceBuilder(
     },
   };
 }
+
+function updateAttribute(
+  {
+    updatedTenant,
+    targetTenant,
+    attributeId,
+    revocationTimestamp,
+  }: {
+    updatedTenant: Tenant;
+    targetTenant: WithMetadata<Tenant>;
+    attributeId: AttributeId;
+    revocationTimestamp?: Date | undefined;
+  },
+  assignmentTimestamp: Date = new Date()
+): Tenant {
+  return {
+    ...updatedTenant,
+    attributes: targetTenant.data.attributes.map((a) =>
+      a.id === attributeId
+        ? {
+            ...a,
+            assignmentTimestamp,
+            revocationTimestamp,
+          }
+        : a
+    ),
+  };
+}
```

Comment:
I think we should avoid to have this function which handles both assignment and revocation (especially because `revocationTimestamp` is optional and `assignmentTimestamp` has a fallback value). If we don't want to split the function  in two, we could just remove it and put this logic inside the higher level, when it's needed.

Reply:
I thought so too, but it's a code repetition that happens in almost all routes, that's why `updateAttribute` was created, let's see what @AsterITA and @Carminepo2 also think

Reply:
Since this logic is used frequently, it would be beneficial to maintain it as a separate function.

`updateAttribute` is agnostic to what it should update; it simply applies the input it receives. The endpoint that utilizes it has the responsibility of determining what to update based on its specific logic, so I don't believe this poses an issue.

The optional `assignmentTimestamp` parameter can be useful as it typically it's `new Date()`, with the exception of a specific instance if I recall correctly. However, we can make it mandatory if it causes confusion.

Reply:
The problem is that assignment and revocation are completely opposite operations, so we should avoid unclear situations in any ways. cc @galales

Reply:
I'm with @taglioni-r on this one.

Reply:
Ok, let's wait to see what @galales says and if so I'll change everything

Reply:
> Ok, let's wait to see what @galales says and if so I'll change everything

IMHO you can go ahead and remove it in this case